core/src/main/scala/kafka/server/DynamicBrokerConfig.scala

```diff
@@ -535,8 +538,8 @@ class DynamicBrokerConfig(private val kafkaConfig: KafkaConfig) extends Logging
     if (changeMap.nonEmpty || deletedKeySet.nonEmpty) {
       try {
         val customConfigs = new util.HashMap[String, Object](newConfig.originalsFromThisConfig) // non-Kafka configs
-        newConfig.valuesFromThisConfig.keySet.forEach(customConfigs.remove(_))
-        reconfigurables.foreach {
+        newConfig.valuesFromThisConfig.keySet.forEach(k => customConfigs.remove(k))
+        reconfigurables.asScala.foreach {
```

Comment:
nit: Could we use `forEach` instead of `asScala.foreach` here? I am not sure if it works though.